### PR TITLE
App store mdx support

### LINKED ISSUE #2291: screenshots for app details
- max 5
- use the same slider as "popular apps" slider 
- use nextjs Image
- screenshots in app folder
- maybe 1.jpg, 2.jpg, ..?

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Fixes #2291

- Moves all static assets to their respective App Store folders
- Creates an endpoint to serve App Store static assets from outside the `/public` directory
- Adds MDX support for App's READMEs
- Adds `gray-matter` support for MDX as well.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- [ ] Update one of the `README.mdx` on any app of the app store
- [ ] Se the changes in `/apps/[slug]`

### CODE REVIEW

**Review by pumfleet — COMMENTED** [review]
Code looks good to me, will test properly soon

File: apps/web/pages/apps/[slug].tsx [hunk 1]
```diff
@@ -1,12 +1,49 @@
...
+      <Slider<string>
+        items={items}
+        title="Screenshots"
+        options={{
+          perView: 1,
+        }}
+        renderItem={(item) =>
+          isTabletAndUp ? (
+            <Image src={item} alt="" layout="fixed" width={573} height={382} />
+          ) : (
```
Comment (hariombalhara) [inline apps/web/pages/apps/[slug].tsx:37]: There is this weird behaviour with lazily loaded images. Setting loading="eager" fixes the issue.
https://www.loom.com/share/72426c9efe1a4d55bd9c7e69fb60aa4f

**Review by hariombalhara — APPROVED** [review]
Looks good. Apart from 2 non blocker issues I mentioned.

File: apps/web/pages/api/app-store/[...static].ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * ```text
+ * Requesting: `/api/app-store/zoomvideo/icon.svg` from a public URL should
+ * serve us the file located at: `/packages/app-store/zoomvideo/static/icon.svg`
+ * ```
+ * This will allow us to keep all app-specific static assets in the same directory.
+ */
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const [appName, fileName] = Array.isArray(req.query.static) ? req.query.static : [req.query.static];
+  const fileNameParts = fileName.split(".");
+  const { [fileNameParts.length - 1]: fileExtentsion } = fileNameParts;
```
[the remark this answers is not in the document]
Comment (zomars) [reply apps/web/pages/api/app-store/[...static].ts:19]: nice catch

**Pull request merged**